api/hypershift/v1beta1/azure.go

```diff
@@ -478,6 +482,17 @@ type ManagedIdentity struct {
 	//
 	// +kubebuilder:validation:Required
 	CertificateName string `json:"certificateName"`
+
+	// objectEncoding represents the encoding for the Azure Key Vault secret containing the certificate related to
+	// CertificateName. objectEncoding needs to match the encoding format used when the certificate was stored in the
+	// Azure Key Vault. If objectEncoding doesn't match the encoding format of the certificate, the certificate will
+	// unsuccessfully be read by the Secrets CSI driver and an error will occur.
```

Comment:
can we document what's the default?
will this error be signaled back in any way to the api consumer?

Reply:
Updated

Reply:
Also this API change is only for M2. This will not be needed for M3 and will be removed when we start the MI phase 3 work.